pkg/cloud/services/compute/instance.go

```diff
@@ -265,10 +266,17 @@ func createInstance(is *Service, clusterName string, i *infrav1.Instance) (*infr
 
 	serverCreateOpts = applyServerGroupID(serverCreateOpts, i.ServerGroupID)
 
+	mc := metrics.NewMetricPrometheusContext("server", "create")
+
 	server, err := servers.Create(is.computeClient, keypairs.CreateOptsExt{
 		CreateOptsBuilder: serverCreateOpts,
 		KeyName:           i.SSHKeyName,
 	}).Extract()
+
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+
 	if err != nil {
```

Comment:
```suggestion
	if mc.ObserveRequest(err) != nil {
```

ObserveRequest just exposes a metric and does return the very same error.